src/core/rooms.ts

```diff
@@ -44,15 +54,40 @@ export interface Rooms {
   get clientOptions(): ClientOptions;
 }
 
+/**
+ * Represents an entry in the chat room map.
+ */
+interface RoomMapEntry {
+  /**
+   * The promise that will eventually resolve to the room.
+   */
+  promise: Promise<DefaultRoom>;
+
+  /**
+   * A random, internal identifier useful for debugging and logging.
+   */
+  nonce: string;
+
+  /**
+   * The options for the room.
+   */
+  options: RoomOptions;
+
+  /**
+   * An abort controller to abort the get operation if the room is released before the get operation completes.
+   */
+  abort?: AbortController;
+}
+
 /**
  * Manages the chat rooms.
  */
 export class DefaultRooms implements Rooms {
   private readonly _realtime: Ably.Realtime;
   private readonly _chatApi: ChatApi;
   private readonly _clientOptions: NormalizedClientOptions;
-  private readonly _rooms: Map<string, DefaultRoom> = new Map<string, DefaultRoom>();
-  private readonly _releasing = new Map<string, { count: number; promise: Promise<void> }>();
+  private readonly _rooms: Map<string, RoomMapEntry> = new Map<string, RoomMapEntry>();
+  private readonly _releasing = new Map<string, { promise: Promise<void> }>();
```

Comment:
can we do type alias for `Promise<void>`

Reply:
I think this makes things more confusing.

If I see a `Promise<void>` I know exactly what I'm dealing with - a standard, well-understood JS type. If I see anything else, I have to go and look up exactly what it is and it's `thenable` behaviour won't necessarily be obvious to me.